elastic-search-support/src/main/java/uk/gov/hmcts/ccd/definition/store/elastic/ElasticDefinitionImportListener.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.ccd.definition.store.elastic;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.ccd.definition.store.elastic.client.CCDElasticClient;
+import uk.gov.hmcts.ccd.definition.store.elastic.config.CcdElasticSearchProperties;
+import uk.gov.hmcts.ccd.definition.store.elastic.exception.ElasticSearchInitialisationException;
+import uk.gov.hmcts.ccd.definition.store.elastic.mapping.CaseMappingGenerator;
+import uk.gov.hmcts.ccd.definition.store.event.DefinitionImportedEvent;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+
+import java.io.IOException;
+import java.util.List;
+
+@Slf4j
+public abstract class ElasticDefinitionImportListener {
+
+    @Autowired
```

Comment:
just read your next comment and realised that you were referring to autowiring in the subclasses rather than declaring the dependencies in the subclasses
ok done :)

Reply:
Yeah, declaring fields in superclass is fine, i was just referring to wiring :-)